locust_benchmark/README.md

```diff
@@ -0,0 +1,178 @@
+    "schema_repetitions": 1,
+    "recreate_feature_group": true,
+    "batch_size": 100
+}
+```
+
+- `host`: Domain name of your Hopsworks cluster.
+- `port`: optional, for managed.hopsworks.ai it should be `443`.
+- `project`: the project to run the benchmark in.
+- `external`: if you are not running the benchmark suite from the same VPC as Hopsworks, set this to `true`.
+- `rows`: Number of rows/unique primary key values in the feature group used for lookup benchmarking.
+- `schema_repetitions`: This controls the number of features for the lookup. One schema repetition will result in 10 features plus primary key. Five repetitions will result in 50 features plus primary key.
+- `recreate_feature_group`: This controls if the previous feature group should be dropped and recreated. Set this to true when rerunning the benchmark with different size of rows or schema repetitions.
+- `batch_size`: This is relevant for the actual benchmark and controls how many feature vectors are looked up in the batch benchmark.
+
+3. Create the feature group
+
+```bash
+python create_feature_group.py
+```
+
+Note, the `recreate_feature_group` only matters if you rerun the `create_feature_group.py` script.
+
+#### Run one process locust benchmark
+
+You are now ready to run the load test:
+```bash
+locust -f locustfile.py --headless -u 4 -r 1 -t 30 -s 1 --html=result.html
+```
+
+Options:
+- `u`: number of users sharing the python process
+- `r`: spawn rate of the users in seconds, it will launch one user every `r` seconds until there are `u` users
+- `t`: total time to run the test for
+- `s`: shutdown timeout, no need to be changed
+- `html`: path for the output file
+
+You can also run only single feature vector or batch lookups by running only the respective user class:
+
+```bash
+locust -f locustfile.py FeatureVectorLookup --headless -u 4 -r 1 -t 30 -s 1 --html=result.html
+
+locust -f locustfile.py FeatureVectorBatchLookup --headless -u 4 -r 1 -t 30 -s 1 --html=result.html
+```
+
+### Distributed Locust Benchmark using Docker Compose
+
+As you will see already with 4 users, the single core tends to be saturated and locust will print a warning.
+
+
+
+```
+[2023-03-09 12:02:11,304] ip-10-0-0-187/WARNING/root: CPU usage above 90%! This may constrain your throughput and may even give inconsistent response time measurements! See https://docs.locust.io/en/stable/running-distributed.html for how to distribute the load over multiple CPU cores or machines
+```
+
+First we need to install docker, for which we will use [the provided convenience script](https://docs.docker.com/engine/install/ubuntu/#install-using-the-convenience-script), however, you can use any other method too:
+```bash
+curl -fsSL https://get.docker.com -o get-docker.sh
+sudo sh get-docker.sh
+```
+
+#### Load the docker image
+
+```bash
+wget https://repo.hops.works/dev/moritz/locust_benchmark/locust-2.15.0-hsfs-3.2.0.dev1-amd64.tgz
+sudo docker load < locust-2.15.0-hsfs-3.2.0.dev1-amd64.tgz
```

Comment:
Why not just use www.dockerhub.com/u/hopsworks? Just refer to `hopsworks/locust:2.15` in the docker-compose file, nothing needs to be pulled/downloaded manually then.